stl/inc/future

```diff
@@ -272,6 +288,8 @@ public:
             _Rethrow_future_exception(_Exception);
         }
 
+        // TRANSITION: This should be assigned before `_Exception` is rethrown so that a `future::get`
+        // that throws a stored exception is invalidated (see: N4901 [futures.unique.future]/17)

```

Comment:
I think this could be a bit clearer (and who doesn't want to invalidate the future?)
```suggestion
        // TRANSITION: _Retrieved should be assigned before `_Exception` is thrown so that a `future::get`
        // that throws a stored exception invalidates the future (N4901 [futures.unique.future]/17)
```